drift.py

```diff
@@ -1,146 +1,90 @@
 
     def analyze(self, commit):
-        for spec in self.specs:
-            out = spack("spec --json " + spec)
-            verToken = spec.find("@")
-            if verToken < 0: verToken = len(spec)
-            name = spec[:verToken]
-            result = Result()
-            result.commit = commit.hash
-            result.abstract = name
-            result.timestamp = str(commit.author_date)
-            if len(spec[verToken:]) > 0:
-                result.version = spec[verToken+1:]
-            if out.returncode == SUCCESS:
-                concrete_spec = json.loads(out.stdout)["spec"]
-                # In the case that the concretized spec hash changes
-                # we want to record the commit where this happens.
-                if self.last[spec] != "" and self.last[spec][0][name]["full_hash"] != concrete_spec[0][name]["full_hash"]:
-                    # Check if the package version is different.
-                    if self.last[spec][0][name]["version"] != concrete_spec[0][name]["version"]:
-                        result.tags += ["pkg-updated"]
-                    # +---------------------+
-                    # |     Dependencies    |
-                    # +---------------------+
-                    result.tags += set(self.tag_deps(concrete_spec, spec, name))
-                    # +---------------------+
-                    # |      Variants       |
-                    # +---------------------+
-                    result.tags += set(self.tag_variants(concrete_spec, spec, name))
-                    # Send Inflection Point
-                    send(result)
-                # Save concrete spec hash as last hash.
-                self.last[spec] = concrete_spec
-            else:
-                # If the spec doesn't concretize properly we also want
-                # to record the commit at which this occurred.
-                result.tags += ["concretizaton-failed"]
+        for abstract_spec in self.specs:
+            # Separate out spec version and name.
+            name = abstract_spec
+            version = ""
+            if "@" in abstract_spec:
+                name, version = abstract_spec.split("@", 1)
+            # Check that version exists in package.py file.
+            out = run(f"spack versions --safe-only {abstract_spec}")
+            # Don't attempt to concretize if the version doesn't yet exist.
+            if version in out.stdout:
+                out = run(f"spack spec --yaml {abstract_spec}")
+                # If concretization successful check the resulting concrete specs.
+                if out.returncode == SUCCESS:
+                    concrete_spec = spack.spec.Spec().from_yaml(out.stdout)
+                    # If the spec is the same move on.
+                    if abstract_spec not in self.last:
+                        self.last[abstract_spec] = concrete_spec
+                    if self.last[abstract_spec] == concrete_spec:
+                        continue
```

Comment:
Doesn't this fail to build the first time it inserts a spec?  You do the check after you insert the `concrete_spec`, but you should build it the first time, when there is no last but you've just inserted `concrete_spec`.

So, unless I am missing something:

```python
if self.last.get(abstract_spec) == concrete_spec:
    continue
else:
    self.last[abstract_spec] = concrete_spec
```

I guess the question is whether you care whether the first version built at all.  IMO we do care -- we want to know when working things stopped working.

Reply:
No you're exactly right this isn't marking the first point seen as an inflection point. I guess I had done this by design thinking that we wanted this database to only contain inflection points and without knowing the commit before the starting commit we can't tell if that commit is an inflection point. I'm happy to switch it out however so that the commit we start on is always submitted to the DB as a starting control point/baseline to know if it was building at the beginning of the analysis or not.

Reply:
Sorry I think I misunderstood but @tgamblin do you think we should save the first point checked or not?

Reply:
I think it's useful to know that the first spec actually built (or that it didn't), so yeah I think we should try to check it.  Does that work for you?

Reply:
Yeah this works!